libsolutil/JSON.cpp

```diff
@@ -87,8 +87,31 @@ bool parse(Json::CharReaderBuilder& _builder, string const& _input, Json::Value&
 	return reader->parse(_input.c_str(), _input.c_str() + _input.length(), &_json, _errs);
 }
 
+/// Takes a JSON value (@ _json) and removes all its members with value 'null' recursively.
+void removeNullMembersHelper(Json::Value& _json)
+{
+	if (_json.type() == Json::ValueType::arrayValue)
+		for (auto& child: _json)
+			removeNullMembersHelper(child);
+	else if (_json.type() == Json::ValueType::objectValue)
+		for (auto const& key: _json.getMemberNames())
+		{
+			Json::Value& value = _json[key];
+			if (value.isNull())
+				_json.removeMember(key);
+			else
+				removeNullMembersHelper(value);
+		}
+}
+
 } // end anonymous namespace
 
+Json::Value removeNullMembers(Json::Value _json)
```

Comment:
Does this perform a deep copy?

Reply:
@chriseth There are 7 calls to this. 4 of them are rvalues, and 3 are references. For references, a deep copy is performed.

Reply:
It is possible to avoid the three deep copies for `m_compiler->storageLayout(_contract)`, `m_compiler->natspecDev(_contract)` (similarly user) and  `m_compiler->assemblyJSON(contract)` if these values are also modified. I'm not sure if I should do it. What do you think?

Reply:
On second thought, why not? I'll avoid the deep copies.

Reply:
Actually I think it is best to create a copy but move the non-affected members.

Reply:
... which mainly boils down to the current code I believe :)

Reply:
Ah, I just realized that `m_compiler->storageLayout(...)` etc returns a const reference. So those deep copies cannot be avoided by creating `removeNullMembers(Json::Value& _json)` and `removeNullmembers(Json::Value&& _json)`